File: SITA/SITA2/models/Defense.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Defense = sequelize.define('Defense', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  seminar_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'seminar_submissions',
      key: 'id'
    }
  },
  student_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'users',
      key: 'id'
    }
  },
  title: {
    type: DataTypes.STRING(255),
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  research_object: {
    type: DataTypes.TEXT,
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  methodology: {
    type: DataTypes.TEXT,
    allowNull: false,
    validate: {
      notEmpty: true
    }
  },
  defense_date: {
    type: DataTypes.DATE,
    allowNull: true
  },
  approval_letter_file: {
    type: DataTypes.STRING(255),
    allowNull: true
  },
  status: {
    type: DataTypes.ENUM('pending', 'approved', 'rejected'),
    defaultValue: 'pending'
  },
  rejection_reason: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  suggested_date: {
    type: DataTypes.DATE,
    allowNull: true
  }
}, {
  tableName: 'defense_submissions',
  timestamps: true,
  underscored: true
});

module.exports = Defense;

File: app/src/main/java/com/example/sitaa/model/Defense.kt
package com.example.sitaa.model

import com.google.gson.annotations.SerializedName
import java.util.Date

data class Defense(
    val id: Int,
    @SerializedName("seminar_id")
    val seminarId: Int,
    @SerializedName("student_id")
    val studentId: Int,
    @SerializedName("defense_date")
    val defenseDate: Date?,
    @SerializedName("approval_letter_file")
    val approvalLetterFile: String?,
    val status: String,
    @SerializedName("rejection_reason")
    val rejectionReason: String?,
    @SerializedName("suggested_date")
    val suggestedDate: Date?,
    @SerializedName("created_at")
    val createdAt: Date,
    @SerializedName("updated_at")
    val updatedAt: Date,
    val student: Student
) {
    data class Student(
        val id: Int,
        val name: String,
        val nim: String,
        @SerializedName("profile_photo")
        val profilePhoto: String?
    )

    enum class Status {
        @SerializedName("pending")
        PENDING,
        @SerializedName("approved")
        APPROVED,
        @SerializedName("rejected")
        REJECTED
    }
}